### PR TITLE
LLM-Assisted Translations Frontend (rebased again)

### PR DESCRIPTION
**Pull request by mathjazz**
This is a continuation of https://github.com/mozilla/pontoon/pull/3159, which was easier to rebase by squashing commits preceeding my changes.

### CODE REVIEW

File: translate/src/modules/machinery/components/MachineryTranslation.tsx [hunk 1]
```diff
@@ -80,11 +76,7 @@ export function MachineryTranslationComponent({
       <li
         className={className}
         title='Copy Into Translation (Ctrl + Shift + Down)'
-        onClick={
-          llmTranslation
-            ? copyLLMTranslationIntoEditor
-            : copyRegularTranslationIntoEditor
-        }
+        onClick={() => copyTranslationIntoEditor()}
```
Comment (eemeli) [inline translate/src/modules/machinery/components/MachineryTranslation.tsx:79]:
```suggestion
        onClick={copyTranslationIntoEditor}
```

File: translate/src/modules/machinery/components/source/GoogleTranslation.tsx [hunk 2]
```diff
@@ -6,16 +6,31 @@ import { logUXAction } from '~/api/uxaction';
...
 }: Props): React.ReactElement<'li'> {
+  const TITLE = (
+    <Localized id='machinery-GoogleTranslation--translation-source'>
+      <span className='translation-source'>GOOGLE TRANSLATE</span>
+    </Localized>
+  );
+
+  if (!isOpenAIChatGPTSupported) {
+    return <li className='google-translation'>{TITLE}</li>;
+  }
```
Comment (eemeli) [inline translate/src/modules/machinery/components/source/GoogleTranslation.tsx:32]: This early return breaks the rules of hooks. Also, why capitalize `TITLE`?

File: translate/src/modules/machinery/components/source/GoogleTranslation.tsx [hunk 3]
```diff
@@ -45,42 +60,54 @@ export function GoogleTranslation({
...
-  let selectedOptionText: string;
+  let selectedOptionElement;
   switch (selectedOption) {
     case 'alternative':
-      selectedOptionText = 'REPHRASED';
+      selectedOptionElement = (
+        <Localized id='machinery-GoogleTranslation--option-rephrase-selected'>
+          <span className='selected-option'>REPHRASED</span>
+        </Localized>
+      );
```
Comment (eemeli) [inline translate/src/modules/machinery/components/source/GoogleTranslation.tsx:70]: This switch is unnecessarily verbose. Just set an `l10nId` value in each case, and leave out the fallback that'll be filled out by Fluent in any case. Then the element itself can be defined normally in the return value.

**Review by eemeli — APPROVED** [review]
Some small stuff inline, overall looks fine.

**Review by eemeli — APPROVED** [review]

File: translate/src/modules/machinery/components/source/GoogleTranslation.tsx [hunk 4]
```diff
@@ -66,11 +66,15 @@ export function GoogleTranslation({
         >
           {title}
 
-          <Localized
-            id={`machinery-GoogleTranslation--option-${selectedOption}`}
-          >
+          {selectedOption && selectedOption !== '' ? (
```
Comment (eemeli) [inline translate/src/modules/machinery/components/source/GoogleTranslation.tsx:69]: In JS, an empty string is falsy, so the second condition is unnecessary here.

**Review by eemeli — APPROVED** [review]
Eh, I'd consider the previous ✅ as still valid. 😇

**Review by ayanaar — APPROVED** [review]

File: translate/src/modules/machinery/components/MachineryTranslation.tsx [hunk 5]
```diff
@@ -41,20 +41,16 @@ export function MachineryTranslationComponent({
 
   const { llmTranslation } = useLLMTranslation(translation);
 
-  const copyRegularTranslationIntoEditor = useCallback(() => {
+  const copyTranslationIntoEditor = useCallback(() => {
```
[the remark this answers is not in the document]
Comment (eemeli) [reply translate/src/modules/machinery/components/MachineryTranslation.tsx:44]: The previous iteration (7dbba69) also included an argument for this callback, and I don't feel like making a fuss about this. 😇

**Pull request merged**